util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java

```diff
@@ -265,10 +272,25 @@ public void write(byte[] b, int offset, int length) throws IOException {
         int bufferSize = Math.min(remaining, 15 * 1024 * 1024);
         byte[] buffer = new byte[bufferSize + 1];
         buffer[0] = stream;
+
         System.arraycopy(b, offset + bytesWritten, buffer, 1, bufferSize);
-        if (!WebSocketStreamHandler.this.socket.send(ByteString.of(buffer))) {
+        ByteString byteString = ByteString.of(buffer);
+
+        int attempts = 0;
+        while (WebSocketStreamHandler.this.socket.queueSize() + byteString.size() > MAX_QUEUE_SIZE
+            && attempts < MAX_QUEUE_SIZE_MAX_ATTEMPTS) {
+          try {
+            Thread.sleep(MAX_QUEUE_SIZE_WAIT_MILLISECONDS);
```

Comment:
Rather than just straight sleeping, what do you think about using `wait(MAX_TIME)/notify` here. It would reduce latency and also be more efficient.

Reply:
Great suggestion! it is actually way long better 

#### Using wait (~3min)

```
# exec 1
2022-09-27T14:12:16.502+0200 [    main][DEBUG] Copying /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/12395450492776725845 to default:tests-c58fcf48c-n5w8f:/tmp
2022-09-27T14:15:44.043+0200 [    main][DEBUG] Copied /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/12395450492776725845

# exec 2
2022-09-27T14:27:31.007+0200 [    main][DEBUG] Copying /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/8625417876961582076 to default:tests-c58fcf48c-n5w8f:/tmp
2022-09-27T14:30:51.087+0200 [    main][DEBUG] Copied /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/8625417876961582076
```

#### Using Thread.sleep (~4min)
```
# exec 1
2022-09-27T14:38:52.833+0200 [    main][DEBUG] Copying /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/422416366883411235 to default:tests-c58fcf48c-n5w8f:/tmp
2022-09-27T14:42:23.119+0200 [    main][DEBUG] Copied /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/422416366883411235

# exec 2
2022-09-27T15:24:35.732+0200 [    main][DEBUG] Copying /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/2455455714318474992 to default:tests-c58fcf48c-n5w8f:/tmp
2022-09-27T15:28:35.794+0200 [    main][DEBUG] Copied /var/folders/v_/9khlk6_94hg0q1prl51fxbkh0000gp/T/tests-workplace-2455455714318474992
```